R/render.R

```diff
@@ -44,7 +47,11 @@ build_site = function(
   if (local && length(files)) {
     files = getOption('blogdown.files_filter', timestamp_filter)(files)
   }
-  build_rmds(files)
```

Comment:
I thought a while about it and I tend to let users decide how to parallelize it because there are multiple ways of parallelization. Here is how I'd implement it:

```r
build_fun = getOption('blogdown.build_rmds', function(files, build) {
  build(files)
})
build_fun(files, build_rmds)
```

Then users can set their own functions ([in `.Rprofile`](https://bookdown.org/yihui/blogdown/global-options.html)), e.g.

```r
options(blogdown.build_rmds = function(files, build) {
  parallel::mclapply(files, build)
})
```

Reply:
- Updated the approach to hopefully be in line with your though process, albeit still exposing the `build_fun` as a parameter, not just retrieving it from options within the function. We can also remove the parameter if you like the other approach better.

- Would suggest to keep the `build_rmds_parallel()`, as it gives users an easy way to get parallelization without having to implement themselves, using either

  `options("blogdown.build_rmds" = blogdown:::build_rmds_parallel)`

  or (assuming we keep the parameter) calling directly

  `blogdown::build_site(build_fun = blogdown:::build_rmds_parallel)`